core-api/src/test/java/com/optimizely/ab/event/internal/EventFactoryTest.java

```diff
@@ -249,6 +244,56 @@ public void createImpressionEventIgnoresUnknownAttributes() throws Exception {
         }
     }
 
+
+    /**
+     * Verify that passing through an list value attribute causes that attribute to be ignored, rather than
+     * causing an exception to be thrown.
+     */
+    @Test
+    public void createImpressionEventIgnoresInvalidAttributes() {
+        // use the "valid" project config and its associated experiment, variation, and attributes
+        ProjectConfig projectConfig = validProjectConfig;
+        Experiment activatedExperiment = projectConfig.getExperiments().get(0);
+        Variation bucketedVariation = activatedExperiment.getVariations().get(0);
+        Attribute attribute = validProjectConfig.getAttributes().get(0);
+        List invalidAttribute = new LinkedList();
+        LogEvent impressionEvent =
+                factory.createImpressionEvent(projectConfig, activatedExperiment, bucketedVariation, "userId",
+                        Collections.singletonMap(attribute.getKey(), invalidAttribute));
+
+        EventBatch impression = gson.fromJson(impressionEvent.getBody(), EventBatch.class);
+
+        // verify that no Feature is created for attribute.getKey() -> invalidAttribute
+        for (com.optimizely.ab.event.internal.payload.Attribute feature : impression.getVisitors().get(0).getAttributes()) {
+            assertNotSame(feature.getKey(), attribute.getKey());
+            assertNotSame(feature.getValue(),  invalidAttribute);
+        }
+    }
+
+    /**
+     * Verify that passing through an null value attribute causes that attribute to be ignored, rather than
+     * causing an exception to be thrown.
+     */
+    public void createImpressionEventIgnoresNullAttributes() {
+        // use the "valid" project config and its associated experiment, variation, and attributes
+        ProjectConfig projectConfig = validProjectConfig;
+        Experiment activatedExperiment = projectConfig.getExperiments().get(0);
+        Variation bucketedVariation = activatedExperiment.getVariations().get(0);
+        Attribute attribute = validProjectConfig.getAttributes().get(0);
+
+        LogEvent impressionEvent =
+                factory.createImpressionEvent(projectConfig, activatedExperiment, bucketedVariation, "userId",
+                        Collections.singletonMap(attribute.getKey(), null));
+
+        EventBatch impression = gson.fromJson(impressionEvent.getBody(), EventBatch.class);
+
+        // verify that no Feature is created for attribute.getKey() -> null
+        for (com.optimizely.ab.event.internal.payload.Attribute feature : impression.getVisitors().get(0).getAttributes()) {
+            assertNotSame(feature.getKey(), attribute.getKey());
+            assertNotSame(feature.getValue(),  null);
+        }
+    }
+
```

Comment:
Can we include explicit unit tests to make sure 
* Integers and Decimals are both tracked. I'm not sure if we can test them distinctly via the compatibility suite, since attribute values have to be serialized via JSON.
* Longs, BigIntegers, and BigDecimals aren't tracked? We definitely can't test these via the compatibility suite.

Maybe test that Booleans are also trackable (in addition to Strings, which I assume are already tested) but I'm less concerned since those can theoretically be covered by the compatibility suite.